IPs/RSKIP175.md

```diff
@@ -0,0 +1,107 @@
+# Fast Bridge
+|RSKIP          |           |
+| :------------ |:-------------|
+|**Title**      |Fast Bridge |
+|**Created**    |15-SEP-20 |
+|**Author**     |GM |
+|**Purpose**    |USa |
+|**Layer**      |Core |
+|**Complexity** |2 |
+|**Status**     |Draft |
+
+## Abstract
+
+This RSKIP allows a user to transfer BTC to RSK in a fast way, where a third party takes the risk to advance the payment in RBTC to the user.
+
+## Motivation
+
+Allowing users to transfer BTC to RSK immediately, without waiting for usual peg-in confirmations. This also generates a market between user who want to fast peg-in and those who count on liquidity.
+
+## Specification
+
+The system comprises one or more liquidity providers (LPs), listed in an onchain marketplace. Interactions with LPs are intermediated using a dApp or a website.
+
+#### Liquidity Providers contract
+
+In order to interact with Bridge, the Liquidity Provider contract must have an API format with functions: 
+
+        function transferStatus(bytes32[]) return uint external {
+            return status;
+        }
+
+Where status is an integer value:
+
+**0 = OK**
+**-1 = UNKNOWN
+-2 = ALREADY_PROCESSED
+-3 = FAILED_TO_PROCESS**
+
+And a payable function:
+
+        function transferFunds(bytes32[]) external payable {
+            emit FundsReceived(msg.value);
+        }
+
+
+#### Bridge
+
+The bridge is the most affected by this project. It will include:
+- Create **registerBtcTransfer**() method.
+
+#### registerBtcTransfer()
+
+**Parameters**:
+- *Transaction* rskTx: RSK transaction
+- *byte[]* btcTxSerialized: Serialized Bitcoin transaction
+- *int* height: Block number where the transaction is present
+- *byte[]* pmtSerialized: Serialized partial merkle tree
+- *Sha256Hash* derivationArgumentsHash: A hash created from all the derivation arguments
+- *Address* userRefundAddress: User BTC refund address
+- *RskAddress* LBCAddress: Liquidity Bridge Contract address
+- *Address* LPBtcAddress: Liquidity Provider BTC address
+
+**Flow**:
+
+- Verifies the transaction hash is part of the merkle tree, and that the transactions was not already processed
+- Verifies tx is sending funds to address derivated from Federation using the provided derivation arguments (creating a Hash256 from those arguments)
+- Verifies provided Liquidity Bridge Contract address actually belongs to a contract
+- Get status calling *transferStatus* method from Liquidity Bridge Contract:
+  - If status obtained is **UNKNOWN**, **ALREADY_PROCESSED** or **FAILED_TO_PROCESS**, refund to user is made.
+  - If status obtained is **OK**, will check if Locking Cap value is surpassed:
+    - If not surpassed, will transfer funds to Liquidity Bridge Contract
+	- If surpassed, will refund Liquidity Provider
+
```

Comment:
It seems that there are a few missing steps.
- There should be validation to determine if the UTXO to the derive Fed was already registered.
- There should be a step to register the UTXO